### ISSUE TITLE
Covid Certificate recognizes and loads UK NHS 2D barcodes, but refuses to verify 

### ISSUE THREAD

**Issue by michaelmaguire** [admin-ch/CovidCertificate-SDK-Kotlin#24]
I received a complete 2 dose course of AstraZeneca from the National Health Service (NHS) in the United Kingdom.

I am a Swiss resident. When I try to scan and store that certificate into Swiss Covid Certificate Android app, it successfully reads the certificate details of both doses, but states the certificate is invalid.

What is the trust relationship issue that's not supported? I note that Switzerland recognizes all vaccines approved by SwissMedic, the European Medicines Agency and the World Health Organization. 

My 2 doses of AstraZeneca were already recognized by cantonal authorities as being valid enough to exempt me from quarantine even after travel from an area with a variant of concern.

**Comment by michaelmaguire** [comment]
I get both: "S|TIV, N|TMV"

**Comment by goebelUB** [comment]
As you correctly observe `N|TMV` indicates that there are "too many vaccination entries". `S|TIV` kinda says the same thing, it's somewhat duplicate/there for historical implementation reasons. So what does this mean?

The [current EU DCC schema](https://github.com/ehn-dcc-development/ehn-dcc-schema/blob/release/1.3.0/DCC.combined-schema.json) states that there must only be one entry per certificate (see also [this question](https://github.com/ehn-dcc-development/ehn-dcc-schema/issues/110)). This is reflected in the corresponding [EU regulation](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A32021R0953&qid=1626181554533) (article 3, paragraph 3). This regulation may not apply to Switzerland, but if UK citizens want to travel to the EU this may be an issue. (Again, I never studied law, but I'm interested in these aspects, that's why I go on about them anyway :) )
 
The COVID Cert app should display both entries in the details screen. Does it do that for you?
Nevertheless, it won't accept the certificate as valid since it violates the current EU DCC schema to which Switzerland wants to adhere.
 
You might also want to get in touch with [BAG](mailto:Covid-Zertifikat@bag.admin.ch) directly, I can only speak from a technical point of view. I do feel your problem though, as a German national vaccinated in Switzerland I was awaiting [Swiss-EU acceptance](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A32021D1126&qid=1626182770552).

**Comment by michaelmaguire** [comment]
> The COVID Cert app should display both entries in the details screen. Does it do that for you?

It does display both entries in the details screen.

**Comment by michaelmaguire** [comment]
> The current EU DCC schema states that there must only be one entry per certificate (see also this question). 

Does this mean that the current EU guideline would only show the date of 2nd dose for 2-dose vaccination regimes?

**Comment by michaelmaguire** [comment]
I've also raised this to the NHS App team:

> Thank you for contacting the customer service team at NHS App and NHS login, this is an automated email to say that we have received your message and we will be responding shortly. Your reference number is (120438)

**Comment by goebelUB** [comment]
Correct, each certificate only contains the date of that very vaccination (or test or recovery). Thus without the first certificate you can't tell how far apart in time the two shots were.

**Comment by michaelmaguire** [comment]
> Correct, each certificate only contains the date of that very vaccination (or test or recovery). Thus without the first certificate you can't tell how far apart in time the two shots were.

The EU approach is proving to be a problem for some people, e.g.:

"Just arrived in YVR and although I am fully vaccinated they are making me do a hotel quarantine with my kids because my LU Euronorm vaccination pass only has vaccination information for the second vaccine on it. Fair warning. This was an extremely expensive mistake."

https://www.facebook.com/groups/2011343209196466/permalink/2820736368257142/

**Comment by tilmantroester** [comment]
To add to this, the French trustlist includes the UK public keys and the French Apps (TousAntiCovid and TousAntiCovid Verif) can verify UK QR codes (including the recently added Scottish ones). It's not clear to me if the reason for the French trustlist being the only one to include the UK public keys (going by https://github.com/section42/hcert-trustlist-mirror/) is technical or political.

**Comment by michaelmaguire** [comment]
The Italian Covid 19 checker app now recognizes UK NHS certs as of mid August https://play.google.com/store/apps/details?id=it.ministerodellasalute.verificaC19
